impeller/scene/animation/animation_clip.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "impeller/scene/animation/animation.h"
+
+namespace impeller {
+namespace scene {
+
+class Node;
+class AnimationPlayer;
+
+class AnimationClip final {
+ public:
+  AnimationClip(std::shared_ptr<Animation> animation, Node* bind_target);
+  ~AnimationClip();
+
+  AnimationClip(AnimationClip&&);
+  AnimationClip& operator=(AnimationClip&&);
+
+  bool IsPlaying() const;
+
+  void SetPlaying(bool playing);
+
+  void Play();
+
+  void Pause();
+
+  void Stop();
+
+  bool GetLoop() const;
+
+  void SetLoop(bool looping);
+
+  Scalar GetPlaybackSpeed() const;
+
+  /// @brief  Sets the animation playback speed. Negative values make the clip
+  ///         play in reverse.
+  void SetPlaybackSpeed(Scalar playback_speed);
+
+  Scalar GetWeight() const;
+
+  void SetWeight(Scalar weight);
+
+  /// @brief  Get the current playback time of the animation.
+  Scalar GetPlaybackTime() const;
```

Comment:
I started going down this road but things are getting a little hairy since stuff like the modulo operator is missing for float-backed durations. I'll open a follow-up patch to make the diff clear.